src/sql/session/session.rs

```diff
@@ -130,9 +131,9 @@ impl fmt::Debug for Session {
     }
 }
 
-impl Default for Session {
-    /// Constructs a new `Session` with default values.
-    fn default() -> Session {
+impl Session {
+    /// Given a connection id, provides a new session with default values.
+    pub fn new(conn_id: u32) -> Session {
```

Comment:
should we assert or debug_assert that conn_id is not the dummy connection id?

Reply:
I'd be happy to add that in! Is the concern that the `IdAllocator` may return the dummy connection id in the future?

Also, is there a reason to use `debug_assert` over `assert` here? I would think we would want this assertion to run for release builds as well.

Reply:
Yeah, presumably the concern is that someone refactors the ID allocator without realizing that 0 is special and should not be allocated.

Typically you want to use `debug_assert!` when the assertion is too expensive to run in production. That's not the case here, so agreed that `assert!` is the way to go. 

Arguably `conn_id` should be

```rust
enum {
    System,
    User(u32),
}
```

and then it would be much clearer when you were using the "system" dummy connection—I don't think you'd need assertions in that case. But that's probably a big annoying refactor that I don't particularly recommend at the moment.